sw/device/lib/dif/dif_edn.c

```diff
@@ -68,27 +68,75 @@ dif_result_t dif_edn_set_auto_mode(const dif_edn_t *edn,
   }
   DIF_RETURN_IF_ERROR(check_locked(edn));
 
-  uint32_t reg = mmio_region_read32(edn->base_addr, EDN_CTRL_REG_OFFSET);
-  reg = bitfield_field32_write(reg, EDN_CTRL_CMD_FIFO_RST_FIELD,
-                               kMultiBitBool4True);
-  mmio_region_write32(edn->base_addr, EDN_CTRL_REG_OFFSET, reg);
+  // Check that EDN is disabled.  If it is not disabled (e.g., through a call
+  // to `dif_edn_stop()`), we are not ready to change mode.
+  uint32_t ctrl_reg = mmio_region_read32(edn->base_addr, EDN_CTRL_REG_OFFSET);
+  const uint32_t edn_en =
+      bitfield_field32_read(ctrl_reg, EDN_CTRL_EDN_ENABLE_FIELD);
+  if ((enum multi_bit_bool)edn_en != kMultiBitBool4False) {
```

Comment:
I would use `dif_multi_bit_bool_to_toggle` here.

Reply:
or at the least, make the `edn_en` a `multi_bit_bool_t` type on line 76. (note you could do both)